### ISSUE TITLE
Metrics / Monitoring

### ISSUE THREAD

**Issue by tschoonj** [SciCatProject/catanie#710]
Hi,

We were wondering how the metrics of Scicat are currently being monitored. Can this be done simply with Google Analytics after inserting a tracking ID into the build? 

We have recently set up a Grafana and Prometheus installation: what are your thoughts on adding support for Prometheus into the code (Catamel and/or Catanie) using a client like https://github.com/siimon/prom-client?

Thanks,

Tom

@lashemilt, @N-Lin, @markbasham

**Comment by stephan271** [comment]
Sorry, this is beyond my expertise. May be Henrik knows more ?

**Comment by zjttoefs** [comment]
We have found that for most metrics we can query the database. Like "how many datasets do we store".
That works less well for measuring user interaction. But we don't have a strong use case. 
Some generic information can also come from the orchestration.

Is there anything in particular you would need to know @tschoonj?

**Comment by tschoonj** [comment]
It would be useful to know how often it is used, as well as which features (pages) are more popular than others. This is typically done using Google Analytics but I am not sure how useful this will be in the long run as browsers are starting to block trackers like this.

Of course keeping an eye on the number of datasets would be great as well, but would involve adding (something like) a Prometheus client to the code.

**Comment by zjttoefs** [comment]
In the discussion we had today, there was some resistance to add Google Analytics. If someone wanted to make a pull request to make adding custom headers easier in general, that could be accepted, though.
At ESS we have an investment in graphana for live metrics, so we might favour that over prometheus. But live visualisation of  the number of datasets is lower down our list of interesting features to add. Logging the number of searches or what people search for, might be useful. But could be done straight to mongodb. 
In summary do not hold your breath for any of these feature to appear. Unless there are volunteers.

**Comment by tschoonj** [comment]
We also have a Grafana instance which we use for displaying live metrics. 

However Grafana and Prometheus do not perform the same functions: Grafana needs datasources, and Prometheus is just one possible provider of such data streams, probably the most popular one. To produce the data streams, Prometheus must scrape metrics endpoints, which in this case would have to be built into scicat. If done properly, it should be able to provide pretty much everything that Google Analytics can produce, plus any other metric that support is added for.
